src/config.rs

```diff
@@ -516,11 +496,32 @@ fn parse_dependency(name: impl Into<String>, value: &Item) -> Result<Dependency>
                 }
                 None => None,
             };
+
+            let tag = match table.get("tag").map(|v| v.as_str()) {
+                Some(Some(tag)) => Some(tag.to_string()),
+                Some(None) => {
+                    return Err(ConfigError::InvalidField { field: "tag".to_string(), dep: name });
+                }
+                None => None,
+            };
+
+            let branch = match table.get("branch").map(|v| v.as_str()) {
+                Some(Some(tag)) => Some(tag.to_string()),
+                Some(None) => {
+                    return Err(ConfigError::InvalidField {
+                        field: "branch".to_string(),
+                        dep: name,
+                    });
+                }
+                None => None,
+            };
```

Comment:
There is lacking validation here, what if the user sets the `rev` and the `tag`? We should probably raise an error if more than one is set. See my other comment about an enum type for `rev/tag/branch`.